### PR TITLE
DDF-1457 - Merged cache namespaced commands into catalog commands namespace.

### PR DESCRIPTION
**Pull request by EIrwin**
Removed cache namespace commands and merged remove, removeAll and search functionality into catalog namespace. 

Using the `--cache` option will allow the `catalog:remove,removeAll,search` commands to use the cache store.

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/146)

### CODE REVIEW

**Comment by EIrwin** [comment]
@millerw8 @jlcsmith @kcwire @wmcnalli @beyelerb @emanns95 @kguerriero @shane-voisard @bdeining

@beyelerb is the hero

**Comment by beyelerb** [comment]
Review status: 0 of 18 files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 80 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8Vykakw2fOM0s65Ia)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L80)):</sup>
The description "Use cache" isn't completely clear - does that mean the cache will be cleared in addition to the store? Maybe change the description to "target the cache" or something to let them know that only cache entries will be removed.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 113 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8VUVrvDmBsa8EDTql)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L113)):</sup>
It would probably be good to add a log statement with the same information - helpful when trying to figure out what has happened. The log persists whereas the console message shows up and then disappears. In both this method and the executeRemoveAllFromStore method. Include something about clearing by a console command.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java, line 55 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8WOEWzylRV7tazS_Y)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java#L55)):</sup>
Same comment on the description here.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java, line 60 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8WVaSXNpXQl_gt95q)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java#L60)):</sup>
Add a check for is empty here as well - that has the same result.d

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java, line 80 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8YJ2yZlOJ7ZZev83b)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java#L80)):</sup>
log statements here as well

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SearchCommand.java, line 69 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8Y4r7Nq1d9S4H81Em)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SearchCommand.java#L69)):</sup>
Description here as well

---

<sup>**[catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java, line 44 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8YwLzN_6bKpCF4yL_)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java#L44)):</sup>
It's probably worth adding in a tests for a null list, and a list with more than one element to make sure those are handled correctly.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by EIrwin** [comment]
Review status: 0 of 18 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 80 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy8Vykakw2fOM0s65Ia)** ([raw file](https://github.com/codice/ddf/blob/8a9482ae8d702c0b30081fbc911f87b9f72918df/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L80)):</sup>
Switched this to "Only removed cached entries"

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by andrew-fiedler** [comment]
Review status: 0 of 18 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java, line 56 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy9DqfxXRg-oUOe3XUr)** ([raw file](https://github.com/codice/ddf/blob/22e43f177285c694d496c9a4cf5e0ad6c71e71d8/catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java#L56)):</sup>
You'll want to rename this to tearDownClass.  You can also remove the camel casing on the @Before method name.

https://codice.atlassian.net/wiki/display/DDF/Unit+Testing

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by EIrwin** [comment]
Review status: 0 of 18 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java, line 0 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy9DqfxXRg-oUOe3XUr)** ([raw file](https://github.com/codice/ddf/blob/22e43f177285c694d496c9a4cf5e0ad6c71e71d8/catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java#L0)):</sup>
I will change it, but there is an error in that document. The code example shows 'setUp()' and not 'setup()'

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by shaundmorris** [comment]
Review status: 0 of 18 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java, line 56 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy9DqfxXRg-oUOe3XUr)** ([raw file](https://github.com/codice/ddf/blob/22e43f177285c694d496c9a4cf5e0ad6c71e71d8/catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java#L56)):</sup>
Good catch. I updated the example.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by lessarderic** [comment]
<sup>**[catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java, line 56 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/146#-Jy9DqfxXRg-oUOe3XUr)** ([raw file](https://github.com/codice/ddf/blob/22e43f177285c694d496c9a4cf5e0ad6c71e71d8/catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/TestRemoveCommand.java#L56)):</sup>
And test classes should be named ...Test.java instead of Test...java.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by wmcnalli** [comment]
Reviewed 2 of 18 files at r1, 1 of 4 files at r2, 2 of 3 files at r3.
Review status: 5 of 25 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by jrnorth** [comment]
Reviewed 11 of 18 files at r1, 10 of 10 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 84 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyEQlELcXHA7Kdpo2M9)** ([raw file](https://github.com/codice/ddf/blob/08e640c5d8c66aeb1a27de5f9d26933195608a72/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L84)):</sup>
If there aren't any aliases, you can remove the "aliases" parameter.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 201 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyERuLWK1pVzLb1-514)** ([raw file](https://github.com/codice/ddf/blob/08e640c5d8c66aeb1a27de5f9d26933195608a72/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L201)):</sup>
"remove" should be "removed"

---

<sup>**[catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java, line 38 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyESh55uFVMGKmjRCgX)** ([raw file](https://github.com/codice/ddf/blob/efbde5215be18882f9af3cd49eb5500eb0cec3ad/catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java#L38)):</sup>
The author tag should be removed.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by EIrwin** [comment]
Review status: 21 of 25 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 0 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyERuLWK1pVzLb1-514)** ([raw file](https://github.com/codice/ddf/blob/08e640c5d8c66aeb1a27de5f9d26933195608a72/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L0)):</sup>
This was a typo, it actually says "Only remove cached entries"

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by beyelerb** [comment]
Review status: 21 of 25 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 65 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyFB2Cmo0s4TsomMKW8)** ([raw file](https://github.com/codice/ddf/blob/08e640c5d8c66aeb1a27de5f9d26933195608a72/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L65)):</sup>
Now that the --cache option has been added, this message needs to be more dynamic. Currently you add the --cache option and you are asked if you want to remove all xxx records from the Catalog. If the --cache option is specified, the question should ask if they want the records removed from the cache.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by beyelerb** [comment]
Successfully built and tested functionality in DDF. Looks good. Made a comment about the message presented to the user when removeall with the --cache option is executed. That message should be updated and I'll rebuild and test.

**Comment by beyelerb** [comment]
Rebuilt after last update. The functionality for this ticket looks good. Did notice an issue where deleting an entry from the cache is always successful, whether that entry is actually in the cache (or any entry for that matter) or not. Eric will be writing a separate ticket for that.
Ready to merge after getting thumbs up from other reviewers regarding their comments.

**Comment by jrnorth** [comment]
Reviewed 2 of 4 files at r5.
Review status: 23 of 25 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 201 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyERuLWK1pVzLb1-514)** ([raw file](https://github.com/codice/ddf/blob/08e640c5d8c66aeb1a27de5f9d26933195608a72/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L201)):</sup>
I'm referring to the log message.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by EIrwin** [comment]
Review status: 23 of 25 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java, line 204 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/146#-JyERuLWK1pVzLb1-514)** ([raw file](https://github.com/codice/ddf/blob/08e640c5d8c66aeb1a27de5f9d26933195608a72/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java#L204)):</sup>
Ahhh...I misread this then, This fix has been made now and is committed.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by wmcnalli** [comment]
LGTM

---

Reviewed 1 of 4 files at r5, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Comment by kguerriero** [comment]
Reviewed 13 of 18 files at r1, 1 of 3 files at r3, 7 of 10 files at r4, 3 of 4 files at r5, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

File: catalog/core/catalog-core-commands/pom.xml [hunk 1]
```diff
@@ -82,6 +82,12 @@
             <artifactId>gt-cql</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>ddf.catalog.core</groupId>
+            <artifactId>catalog-core-standardframework</artifactId>
+            <version>${project.parent.version}</version>
```
Comment (shaundmorris) [inline catalog/core/catalog-core-commands/pom.xml]: Just an FYI more than anything else: http://stackoverflow.com/questions/1981151/warning-on-using-project-parent-version-as-the-version-of-a-module-in-maven-3#1982027 ${project.version} might be better. In our case it is all the same version now so shouldn't matter

**Comment by jlcsmith** [comment]
Reviewed 9 of 18 files at r1, 1 of 10 files at r4, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/146)

**Pull request merged**